src/main/groovy/com/gradleup/jaxb/Jaxb2Plugin.groovy

```diff
@@ -71,12 +72,58 @@ class Jaxb2Plugin implements Plugin<Project> {
     // add new tasks for creating/cleaning the auto-value sources dir
     project.task(type: CleanJaxb2SourcesDir, "cleanJaxb2SourcesDir")
     project.task(type: InitJaxb2SourcesDir, "initJaxb2SourcesDir")
-    project.task(type: GenerateJaxb2Classes, "generateJaxb2Classes")
 
     // make 'compileJava' require the new task, so that all sources are available
     project.tasks.clean.dependsOn project.tasks.cleanJaxb2SourcesDir
-    project.tasks.generateJaxb2Classes.dependsOn project.tasks.initJaxb2SourcesDir
-    project.tasks.compileJava.dependsOn project.tasks.generateJaxb2Classes
+
+    project.afterEvaluate {
+      Set<XjcTaskConfig> xjcConfigs = project.extensions.jaxb2.xjc
+      for (XjcTaskConfig xjcConfig : xjcConfigs) {
+        registerGenerateJaxb2ClassesTask(project, xjcConfig)
+      }
+    }
+  }
+
+  private static void registerGenerateJaxb2ClassesTask(Project project, XjcTaskConfig xjcConfig) {
+    def generationTaskConfig = xjcConfig
+    def generationTask = project.tasks.register("generateJaxb2Classes-$generationTaskConfig.name", GenerateJaxb2Classes) {
+
+      it.generatedSourcesDirectory.convention(project.layout.buildDirectory.dir("gradleup/jaxb/${generationTaskConfig.name}"))
+      it.schemaFile.convention(project.layout.projectDirectory.file(generationTaskConfig.schema))
+      it.basePackage.convention(generationTaskConfig.basePackage)
+      it.encoding.convention(generationTaskConfig.encoding)
+      it.extension.convention(generationTaskConfig.extension)
+      it.additionalArgs.convention(generationTaskConfig.additionalArgs)
+      it.header.convention(generationTaskConfig.header)
+
+      if (generationTaskConfig.catalog != null) {
+        it.catalogFile.convention(project.layout.projectDirectory.file(generationTaskConfig.catalog))
+      }
+
+      it.bindingsFiles.from(collectBindingsFiles(project, generationTaskConfig.bindingsDir, generationTaskConfig.includedBindingFiles))
+
+      dependsOn project.tasks.initJaxb2SourcesDir
+    }
+
+    def packagePath = generationTaskConfig.basePackage.replace(".", "/")
+    def processGeneratedClassesTask = project.tasks.register("processJaxb2Classes-$generationTaskConfig.name", Copy) {
```

Comment:
> Similarly, is there a reason the task name contains a hyphen? That is quite unconventional.

Yeah, I did realize this, but I wasn't sure how ill-advised it was. I though it might be the simplest route this way to have the dash as a separator and then what follows exactly match `generationTaskConfig.name`. Otherwise, I guess you'd just capitalize the first character of `generationTaskConfig.name`, and essentially camel-case it. Sounds like that route would be preferred?

Reply:
> Why is this called process and not copy?

Was following the idea of `processResources`, though I suppose that is more than just a simple `Copy` task like this is. I'll update the name.

Reply:
> I guess you'd just capitalize the first character of generationTaskConfig.name

That's what I'd expect.

Reply:
Done